source/docs/guide/src/nonlinear_bitvec.md

```diff
@@ -0,0 +1,287 @@
+        - Mac: `brew install Singular` and set the `VERUS_SINGULAR_PATH` environment variable when running Verus. (e.g. `VERUS_SINGULAR_PATH=/usr/local/bin/Singular`). For more options, see Singular's [OS X installation guide](https://www.singular.uni-kl.de/index.php/singular-download/install-os-x.html). 
+
+        - Debian-based Linux: `apt-get install singular` and set the `VERUS_SINGULAR_PATH` environment variable when running Verus. (e.g. `VERUS_SINGULAR_PATH=/usr/bin/Singular`). For more options, see Singular's [Linux installation guide](https://www.singular.uni-kl.de/index.php/singular-download/install-linuxunix.html).
+
+        - Windows: See Singular's [Windows installation guide](https://www.singular.uni-kl.de/index.php/singular-download/install-windows.html).
+
+2. Compiling Verus with Singular Support
+    - The `integer_ring` functionality is conditionally compiled when the `singular` feature is set.
+      To add this feature, add the `--features singular` flag when you invoke `vargo build` to compile Verus.
+
+
+
+
+### Details/Limitations
+- This can be used only with **int** parameters.
+- Formulas that involve inequalities are not supported.   
+- Division is not supported.
+- Function calls in the formulas are treated as uninterpreted functions.  If a function definition is important for the proof, you should unfold the definition of the function in the proof function's `requires` clause.
+- When using `integer_ring` lemma, the divisor of `%` operator must not be zero. If a divisor can be zero in the ensures clause of the `integer_ring` lemma, the fact in ensures clause will not be available in the callsite.
+
+### Using `integer_ring` as a helper lemma for nonlinear proofs
+As `integer_ring` feature has several limitations, it is not possible to get an arbitary nonlinear property only with `integer_ring` feature. Instead, it is a common pattern to have `by(nonlinear_arith)` function as a main lemma for the desired property, and use `integer_ring` lemma as a helper function.
+
+To work around the lack of support for inequalities and division, you can often write a helper proof discharged with `integer_ring` and use it to prove properties that are not directly supported by `integer_ring`. Furthermore, you can also add additional variables to the formulas. For example, to workaround division, one can introduce `c` where `b = a * c`, instead of `b/a`. Experiences indicate that introducing many additional variables can help with the proof.
+
+#### Example 1: `integer_ring` as a helper lemma to provide facts on modular arithmetic
+In this below `lemma_mod_difference_equal` function, we have four inequalities inside the requires clauses, which cannot be encoded into `integer_ring`. In the ensures clause, we want to prove `y % d - x % d == y - x`. The helper lemma `lemma_mod_difference_equal_helper` simply provides that `y % d - x % d` is equal to `(y - x)` modulo `d`. The rest of the proof is done by `by(nonlinear_arith)`.
```

Comment:
Here and in the next paragraph:
"In this below `lemma_mod_difference_equal` function" -> "In the `lemma_mod_difference_equal` function below,"

Reply:
"is done by" -> "is done via"